File: JavaScript/Exams/JoroTheRabbit.js
function Solve(args) {
    // Reading the input
    var stringsPositions = args[0].split(", ");
    var terrainLength = stringsPositions.length;
    var terrainNumbers = new Array(terrainLength);
    for (var i = 0; i < terrainLength; i++) {
        terrainNumbers[i] = parseInt(stringsPositions[i]);
    }

    // Implementing the logic
    var maxJumps = 0;
    var currentPosition;
   
    for (var startPosition = 0; startPosition < terrainLength; startPosition++) {
        for (var step = 1; step <= terrainLength; step++) {
            currentPosition = startPosition;
            var visited = {};
            var jumps = 0;
            while (!visited[currentPosition]) {
                visited[currentPosition] = true;
                jumps++;
                var lastPosition = currentPosition;
                currentPosition += step;
                if (currentPosition > terrainLength - 1) {
                    currentPosition -= terrainLength;
                }
                if (terrainNumbers[lastPosition] >= terrainNumbers[currentPosition]) {
                    break;
                }
            }
            if (jumps > maxJumps) {
                maxJumps = jumps;
            }
        }
    }
    return maxJumps;
}

File: C# II Projects/CSharpExam2 - Day2/JoroTheRabbit/JoroTheRabbit.cs
using System;

class JoroTheRabbit
{
    static void Main()
    {
        string[] terrainNumbersStrings = Console.ReadLine().Split(new char[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries);
        int terrainLength = terrainNumbersStrings.Length;
        int[] terrainNumbers = new int[terrainLength];
        for (int index = 0; index < terrainLength; index++)
        {
            terrainNumbers[index] = int.Parse(terrainNumbersStrings[index]);
        }

        int maxJumps = 0;
        int currentPosition = 0;
        int currentJumpsCount = 0;
        int nextPosition = 0;
        for (int startPosition = 0; startPosition < terrainLength; startPosition++)
        {
            for (int step = 1; step <= terrainLength; step++)
            {
                currentPosition = startPosition;
                currentJumpsCount = 1;
                nextPosition = (currentPosition + step) % terrainLength;
                while (nextPosition != startPosition && terrainNumbers[currentPosition] < terrainNumbers[nextPosition])
                {
                    currentJumpsCount++;
                    currentPosition = nextPosition;
                    nextPosition = (currentPosition + step) % terrainLength;
                }
                if (currentJumpsCount > maxJumps)
                {
                    maxJumps = currentJumpsCount;
                }
            }
        }

        Console.WriteLine(maxJumps);
    }
}
